### PR TITLE
adds tests for tablet availability boundry conditions

### LINKED ISSUE #3312: OnDemand Follow-on: ITs that rigorously test TabletLocator Tablet boundary conditions
Create ITs that rigorously test the boundry conditions for tablet with different goal states. So for the batch scanner, batch writer, and scanner try interacting with rows right around the tablet boundries for tablets with different goal states. Like the first row and last row in a tablet with a different goal than the tablet next to it.

### PR DESCRIPTION
**Pull request by keith-turner**
fixes #3312

### CODE REVIEW

File: test/src/main/java/org/apache/accumulo/test/functional/TabletAvailabilityIT.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+  private static String row(int i) {
+    return String.format("%010d", i);
+  }
+
+  private static Text rowt(int i) {
+    return new Text(row(i));
+  }
+
+  private static Mutation newMuation(int row, int val) {
```
Comment (dlmarion) [inline test/src/main/java/org/apache/accumulo/test/functional/TabletAvailabilityIT.java:156]:
```suggestion
  private static Mutation newMutation(int row, int val) {
```

**Review by dlmarion — APPROVED** [review]
